### PR TITLE
Add documentation about integrating APIs with security

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

##  opensearch-security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Documentation
 
 
 2. __Github Issue # or road-map entry, if available:__
N/A

 3. __Description of changes:__
Adding a section in the README with instructions on how to onboard/integrate new APIs with security.

 4. __Why these changes are required?__ 
For consistency and faster development of different plugins wanting to integrate new or existing APIs with security.

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
N/A

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 N/A
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)
N/A

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)
N/A

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by davidlago** [comment]
Thanks for the PR @ohltyler. Apologies for not the long delay in the review... especially as we've since updated the README :( Could you please update to move section down right before the Contributing header as git auto-resolving the conflicts is placing it at the very beginning? Thanks!

**Review by davidlago — APPROVED** [review]

**Comment by cliu123** [comment]
Is there any issue tracking this effort to apply these documentation changes consistently on all plugins?

**Comment by ohltyler** [comment]
> 

I don't believe this is documented in any plugins. The idea of adding it here, is so all plugins can have a central place to refer to.

**Review by peternied — DISMISSED** [review]
I think this is a good starting point, lets add it for now and then we can figure out the best mechanism to get this information out.

@ohltyler Could you create a follow up issue in the plugins repo to track centralized practices / tutorial / documentation that would make sense to link this to?

**Comment by ohltyler** [comment]
> I think this is a good starting point, lets add it for now and then we can figure out the best mechanism to get this information out.
> 
> @ohltyler Could you create a follow up issue in the plugins repo to track centralized practices / tutorial / documentation that would make sense to link this to?

I like this idea, created a tracking issue in opensearch-plugins and assigned myself - https://github.com/opensearch-project/opensearch-plugins/issues/146

**Comment by peternied** [comment]
@ohltyler Looks like we need this change to be rebased so it can pass all of our new CI systems, could you do that so we can merge?

**Review by peternied — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -6,16 +6,23 @@
...
-- [Code of Conduct](#code-of-conduct)
-- [Security](#security)
-- [License](#license)
-- [Copyright](#copyright)
+- [OpenSearch Security Plugin](#opensearch-security-plugin)
+  - [Features](#features)
+    - [Encryption](#encryption)
+    - [Authentication](#authentication)
+    - [Access control](#access-control)
+    - [Audit/Compliance logging](#auditcompliance-logging)
```
Comment (cliu123) [inline README.md:14]: `audit/compliance-logging` instead?

Comment (ohltyler) [reply README.md:14]: why? This is auto-generated with `prettier`

Comment (cliu123) [reply README.md:14]: The format `auditcompliance-logging` looks different from other items. If only this format will work, then let's go with it.

**Review by cliu123 — APPROVED** [review]

**Pull request merged**